src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -726,6 +712,30 @@ SpecialType.System_Collections_Generic_IReadOnlyCollection_T or
             }
         }
 
+        internal bool TryBindIEnumerableCollectionInstance(TypeSymbol targetType, SyntaxNode syntax, BindingDiagnosticBag diagnostics, [NotNullWhen(true)] out BoundExpression? result)
+        {
+            if (targetType is NamedTypeSymbol namedType)
+            {
+                var analyzedArguments = AnalyzedArguments.GetInstance();
+                // https://github.com/dotnet/roslyn/issues/68785: Use ctor with `int capacity` when the size is known.
+                result = BindClassCreationExpression(syntax, namedType.Name, syntax, namedType, analyzedArguments, diagnostics);
+                result.WasCompilerGenerated = true;
+                analyzedArguments.Free();
+                return true;
+            }
+            else if (targetType is TypeParameterSymbol typeParameter)
+            {
+                var arguments = AnalyzedArguments.GetInstance();
+                result = BindTypeParameterCreationExpression(syntax, typeParameter, arguments, initializerOpt: null, typeSyntax: syntax, wasTargetTyped: true, diagnostics);
```

Comment:
>BindTypeParameterCreationExpression [](http://example.com/codeflow?start=25&length=35)

Are we testing the conversion error when this method returns an expression with `HasErrors` set? For instance:
```csharp
static T Create() where T : IEnumerable<int> => [];
```